lib/common/components/Login.js

```diff
@@ -1,31 +1,38 @@
 // @flow
 
-import {Component} from 'react'
+import { useAuth0 } from '@auth0/auth0-react'
+// $FlowFixMe useEffect not recognized by flow.
+import { useEffect } from 'react'
+import { browserHistory } from 'react-router'
 
-import * as userActions from '../../manager/actions/user'
-import auth0 from '../user/Auth0Manager'
-
-import type {Props as ContainerProps} from '../containers/Login'
-
-type Props = ContainerProps & {
-  onHide: () => void,
-  push: string => void,
-  receiveTokenAndProfile: typeof userActions.receiveTokenAndProfile,
-  redirectOnSuccess: string
+type Props = {
+  redirectOnSuccess?: string
 }
 
-export default class Auth0 extends Component<Props> {
-  componentDidMount () {
-    // Show lock when component mounts. NOTE: render method returns null.
-    auth0.loginWithLock(this.props)
-  }
-
-  componentWillUnmount () {
-    // Hide lock when component unmounts
-    auth0.hideLock()
-  }
+/**
+ * This component handles the `/login` route by redirecting the browser to the location specified
+ * in the `redirectOnSuccess` prop (or to the previously active URL or `/home` if none is specified).
+ */
+export default function Login ({
+  redirectOnSuccess = window.location.path || '/home'
```

Comment:
should be `window.location.pathname` (or just remove that and let the default be /home)

Reply:
Yeah that window.location was incorrect, but we could remove the component entirely subsequently. (https://github.com/ibi-group/datatools-ui/pull/1032/commits/11e1879d6b3d9cd30be124a8df68fbdc064ba5c7)